cmd/operator-sdk/cli/cli.go

```diff
@@ -42,91 +30,71 @@ import (
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"sigs.k8s.io/kubebuilder/pkg/cli"
+	kbgov1 "sigs.k8s.io/kubebuilder/pkg/plugin/v1"
+	kbgov2 "sigs.k8s.io/kubebuilder/pkg/plugin/v2"
 )
 
-// GetCLIRoot is intended to creeate the base command structure for the OSDK for use in CLI and documentation
-func GetCLIRoot() *cobra.Command {
-	root := &cobra.Command{
-		Use:   "operator-sdk",
-		Short: "An SDK for building operators with ease",
-		PersistentPreRun: func(cmd *cobra.Command, args []string) {
-			if viper.GetBool(flags.VerboseOpt) {
-				if err := projutil.SetGoVerbose(); err != nil {
-					log.Fatalf("Could not set GOFLAGS: (%v)", err)
-				}
-				log.SetLevel(log.DebugLevel)
-				log.Debug("Debug logging is set")
-			}
-			if err := checkGoModulesForCmd(cmd); err != nil {
-				log.Fatal(err)
-			}
-		},
-	}
+var commands = []*cobra.Command{
+	// Once the KB CLI is made the default, add the "new" command as a way to
+	// scaffold the legacy project layout and mark "new" as deprecated
+	// new.NewCmd()
 
-	root.AddCommand(
-		add.NewCmd(),
-		alpha.NewCmd(),
-		build.NewCmd(),
-		bundle.NewCmd(),
-		cleanup.NewCmd(),
-		completion.NewCmd(),
-		execentrypoint.NewCmd(),
-		generate.NewCmd(),
-		migrate.NewCmd(),
-		new.NewCmd(),
-		olm.NewCmd(),
-		printdeps.NewCmd(),
-		run.NewCmd(),
-		scorecard.NewCmd(),
-		test.NewCmd(),
-		version.NewCmd(),
-	)
+	alpha.NewCmd(),
+	bundle.NewCmd(),
+	cleanup.NewCmd(),
+	completion.NewCmd(),
+	olm.NewCmd(),
+	run.NewCmd(),
+	scorecard.NewCmd(),
+	test.NewCmd(),
+	version.NewCmd(),
+	// TODO(hasbro17): add generate csv command after aligning it for kubebuilder layout
+}
 
-	return root
+func Run() error {
+	cli, _ := GetPluginsCLIAndRoot()
+	return cli.Run()
 }
 
-func checkGoModulesForCmd(cmd *cobra.Command) (err error) {
-	// Certain commands are able to be run anywhere or handle this check
-	// differently in their CLI code.
-	if skipCheckForCmd(cmd) {
-		return nil
-	}
-	// Do not perform this check if the project is non-Go, as they will not
-	// be using go modules.
-	if !projutil.IsOperatorGo() {
-		return nil
-	}
-	// Do not perform a go modules check if the working directory is not in
-	// the project root, as some sub-commands might not require project root.
-	// Individual subcommands will perform this check as needed.
-	if err := projutil.CheckProjectRoot(); err != nil {
-		return nil
+// GetPluginsCLIAndRoot returns the plugins based CLI configured to use operator-sdk as the root command
+// This CLI can run kubebuilder commands and certain SDK specific commands that are aligned for
+// the kubebuilder project layout
+func GetPluginsCLIAndRoot() (cli.CLI, *cobra.Command) {
+	c, err := cli.New(
+		cli.WithCommandName("operator-sdk"),
+		cli.WithPlugins(
+			&kbgov1.Plugin{},
```

Comment:
We probably don't want to support kubebuilder v1 projects..

Reply:
Yep. It is deprecated for too long now and the code will be removed. 
We will probably not have the v1.Plugin.